### ISSUE TITLE
MetricPredicate translates to incompatible query in Elasticsearch REST Client

### ISSUE THREAD

**Issue by lorthirk** [eclipse-kapua/kapua#2852]
**Describe the bug**
The current implementation of `MetricPredicate` translate to a query that consider the Datastore Schema. Specifically, an example `MetricPredicate` would translate to JSON as

```
{
    "query": {
        "range" : {
            "temperature" : {
                "gte" : 10,
                "lte" : 20,
            }
        }
    }
}
```

While it should translate to this JSON to support our current schema:

```
{
    "query": {
        "range" : {
            "metrics.temperature.int" : {
                "gte" : 10,
                "lte" : 20,
            }
        }
    }
}
```

This means that filtering for a metric type and/or value would always return 0 entries

**To Reproduce**
1. Perform a REST call to `/{scopeId}/dataMessages` filtering on `metricName`, `metricType`, `metricMin` and `metricMax`

**Expected behavior**
A set of results matching the name, type and values would return

**Version of Kapua**
1.0.9
1.1.2

**Type of deployment**
[x] Local Vagrant deployment
[x] Docker
[x] Openshift (in its variants) 
[x] Others

**Main component affected**
[ ] Console (in case of console please report info on which browser you encountered the problem)
[x] REST API
[ ] Message Broker
[x] - Others

**Additional context**
Basically any component that uses di ES REST client is affected
